symbolic-uplc/src/ZkFold/Symbolic/UPLC/Evaluation.hs

```diff
@@ -253,6 +298,28 @@ applyPoly ctx SndPair [arg] = do
   MaybeValue p <- evalArg ctx arg []
   (_, ExValue v) <- asPair (Symbolic.fromJust p)
   return $ MaybeValue (symMaybe p v)
+applyPoly ctx (BPFList ChooseList) (ct:tt:et:args) = let
+    b0 = applyPoly ctx (BPFList NullList) [ct]
+    bt =   evalArg ctx (AThunk . Right $ b0) []
+  in applyPoly ctx IfThenElse ((AThunk . Right $ bt):tt:et:args)
+applyPoly ctx (BPFList MkCons) (x0:xs0:_) = do
+  MaybeValue x <- evalArg ctx x0 []
+  let l = Symbolic.fromJust x
+  MaybeValue xs <- evalArg ctx xs0 []
+  let ls = Symbolic.fromJust xs
+  return $ MaybeValue (Symbolic.just $ l L..: (fromJust $ cast ls))
+applyPoly ctx (BPFList HeadList) (xs0:_) = do
+  MaybeValue p <- evalArg ctx xs0 []
+  ExList v <- asList (Symbolic.fromJust p)
+  return $ MaybeValue (symMaybe p (L.head v))
+applyPoly ctx (BPFList TailList) (xs0:_) = do
+  MaybeValue p <- evalArg ctx xs0 []
+  ExList v <- asList (Symbolic.fromJust p)
+  return $ MaybeValue (symMaybe p (L.tail v))
+applyPoly ctx (BPFList NullList) (xs0:_) = do
+  MaybeValue p <- evalArg ctx xs0 []
+  ExList v <- asList (Symbolic.fromJust p)
+  return $ MaybeValue (symMaybe p (L.null v))
 applyPoly _ (BPFList _) _ = error "FIXME: UPLC List support"
```

Comment:
This case can be removed? Or are there other list functions which are not supported yet?